clusters/resource_cluster.go

```diff
@@ -234,21 +221,61 @@ func fixInstancePoolChangeIfAny(d *schema.ResourceData, cluster *Cluster) {
 	}
 }
 
+func listClusterConfigChangedKeys(d *schema.ResourceData) []string {
+	changedKeys := []string{}
+	for k := range clusterSchema {
+		// TODO: create a map if we'll add more non-cluster config parameters in the future
+		if k == "library" || k == "is_pinned" {
+			continue
+		}
+		if d.HasChange(k) {
+			changedKeys = append(changedKeys, k)
+		}
+	}
+	return changedKeys
+}
+
 func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
 	clusters := NewClustersAPI(ctx, c)
 	clusterID := d.Id()
 	cluster := Cluster{ClusterID: clusterID}
 	common.DataToStructPointer(d, clusterSchema, &cluster)
 	var clusterInfo ClusterInfo
 	var err error
-	if hasClusterConfigChanged(d) {
+
+	clusterChangedConfigList := listClusterConfigChangedKeys(d)
+
+	if len(clusterChangedConfigList) > 0 {
 		log.Printf("[DEBUG] Cluster state has changed!")
 		if err := cluster.Validate(); err != nil {
 			return err
 		}
 		cluster.ModifyRequestOnInstancePool()
 		fixInstancePoolChangeIfAny(d, &cluster)
-		clusterInfo, err = clusters.Edit(cluster)
```

Comment:
can you do an edit operation first, if there was no change in numWorkers/autoScale? this is more for readability.

Reply:
done